src/rofl/common/crofdpt.cc

```diff
@@ -799,16 +799,21 @@ void crofdpt::group_mod_reset() {
   send_group_mod_message(cauxid(0), ge);
 }
 
-uint32_t crofdpt::send_features_request(const rofl::cauxid &auxid,
-                                        int timeout_in_secs) {
-  uint32_t xid = ++xid_last;
+rofl::crofsock::msg_result_t
+crofdpt::send_features_request(const rofl::cauxid &auxid,
+                                        int timeout_in_secs,
+										uint32_t *xid) {
 
   rofl::openflow::cofmsg *msg = nullptr;
   try {
     msg =
-        new rofl::openflow::cofmsg_features_request(rofchan.get_version(), xid);
+        new rofl::openflow::cofmsg_features_request(rofchan.get_version(), ++xid_last);
 
-    rofchan.send_message(auxid, msg, ctimespec().expire_in(timeout_in_secs));
+    if (xid != nullptr) {
+    	*xid = xid_last;
```

Comment:
This API used to be called from different threads hence xid_last is atomic. With the changes here and in the following *xid might be set with an invalid xid_last. Please keep the local `uint32_t xid = ++xid_last;` to prevent this.